pkg/services/auth/gitlab.go

```diff
@@ -0,0 +1,191 @@
+package auth
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth/internal"
+	"github.com/weaveworks/weave-gitops/pkg/services/auth/types"
+	"io"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+)
+
+const (
+	serverShutdownSuccessMessage = "The authorization flow was completed, you may close your browser and go back to the CLI.\n"
```

Comment:
:thinking: Assuming this is what shows up in the browser, we probably need to have a nicely-rendered page here eventually. Might be do-able in a follow up.

Reply:
Yeah, that would be nice.  Definitely worth a follow-up